rioxarray/raster_writer.py

```diff
@@ -155,7 +155,9 @@ def to_raster(self, xarray_dataarray, tags, windowed, lock, compute, **kwargs):
         **kwargs
             Keyword arguments to pass into writing the raster.
         """
-        dtype = kwargs["dtype"]
```

Comment:
it isn't obvious to me where the default kwargs (dtype) comes from here (from a user's perspective calling this function without any additional arguments `da.rio.to_raster('raster.tif')`

Reply:
Here it is: https://github.com/corteva/rioxarray/blob/6204661cc9a703307207f548365b020a2d6eac59/rioxarray/raster_array.py#L916

Reply:
Unfortunately, with the complex dtype in rasterio, we will need to have a way to get the rasterio and numpy dtype. For the writer, I think that it is best to assume that dtype could be either the rasterio or numpy dtype and automatically determine the numpy/rasterio dtype from it.

Reply:
> Here it is: 

ah. thanks... still learning how all the pieces fit together!